test/fakeintake/aggregator/traceAggregator.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package aggregator
+
+import (
+	"time"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// TracePayload is a payload type for traces
+// It implements PayloadItem
+type TracePayload struct {
+	*pb.AgentPayload
+	collectedTime time.Time
+}
+
+var _ PayloadItem = &TracePayload{}
+
+// name returns the hostname of the agent that produced the payload
+// TracePayload implements PayloadItem
+func (tp *TracePayload) name() string {
+	return tp.HostName
+}
+
+// GetTags is not implemented
+// TracePayload implements PayloadItem
+func (tp *TracePayload) GetTags() []string {
+	return nil
+}
```

Comment:
Would it make sense to return the tags found in the payload here ?
https://github.com/DataDog/datadog-agent/blob/253752538d260369d2ae3f7ddb93a825ffc230d8/pkg/proto/pbgo/trace/agent_payload.pb.go#L37-L38

Reply:
At first I thought about using that field but then I noticed that it's not being used at all in the [trace writer](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/writer/trace.go#L246-L254). It's likely that the field is used in intake/backend after processing since we share the same proto.

There also span-specific tags that are defined on every span in the payload but I didn't want to return them here since they're not common tags for all the payload. I believe in the future when we start writing tests we'll need more advanced searching and filtering capabilities for span tags.